src/ioabc.cpp

```diff
@@ -99,143 +99,62 @@ bool ABCInput::Import(const std::string &abc)
 // parseABC --
 //
 
-void ABCInput::parseABC(std::istream &infile)
+void ABCInput::ParseABC(std::istream &infile)
 {
     // initialize doc
     m_doc->Reset();
     m_doc->SetType(Raw);
 
-    // read file header
-    std::getline(infile, abcLine);
-    while (abcLine[0] != 'X') {
-        if (infile.eof()) {
-            LogError("ABC import: No tune found");
-            exit(0);
-        }
-        if ((abcLine[0] == '%') && (abcLine[1] == '%'))
-            LogWarning("ABC import: Stylesheet directives are ignored");
-        else if (abcLine[1] == ':')
-            readInformationField(abcLine[0], &abcLine[2]);
-        std::getline(infile, abcLine);
-        ++m_lineNum;
-    }
+    Score *score = NULL;
+    Section *section = NULL;
     CreateHeader();
-
     while (!infile.eof()) {
-        while (!(abcLine[0] == 'X' && abcLine[1] == ':') && !infile.eof()) {
-            std::getline(infile, abcLine);
-            ++m_lineNum;
+        std::getline(infile, abcLine);
+        ++m_lineNum;
+        if (abcLine[0] == 'X') {
```

Comment:
Need to test that !abcLine.empty()

Reply:
Correct. Plus we always use `.at()` operator and not `[]`. This applies elsewhere in the importer.